### PR TITLE
feat: Support bitwise aggregate functions

### PR DESCRIPTION
**Pull request by huaxingao**
## Which issue does this PR close?

Closes #.
Support bitwise  aggregate functions such as `BitAnd`, `BitOr`, `BitXor`

### CODE REVIEW

File: spark/src/main/scala/org/apache/comet/serde/QueryPlanSerde.scala [hunk 1]
```diff
@@ -186,6 +186,13 @@ object QueryPlanSerde extends Logging with ShimQueryPlanSerde {
     }
   }
 
+  private def bitwiseAggTypeSupported(dt: DataType): Boolean = {
+    dt match {
+      case _: IntegerType => true
```
Comment (sunchao) [inline spark/src/main/scala/org/apache/comet/serde/QueryPlanSerde.scala:191]: I think Spark supports all `Integral` type as input type which includes other types such as long and short. Maybe we can add them later.

**Review by sunchao — APPROVED** [review]

File: spark/src/main/scala/org/apache/comet/serde/QueryPlanSerde.scala (hunk 1 above)
Comment (huaxingao) [reply spark/src/main/scala/org/apache/comet/serde/QueryPlanSerde.scala]: Added all the other `IntegralType`

File: spark/src/test/scala/org/apache/comet/exec/CometAggregateSuite.scala [hunk 2]
```diff
@@ -947,6 +947,40 @@ class CometAggregateSuite extends CometTestBase with AdaptiveSparkPlanHelper {
...
+
+            checkSparkAnswerAndNumOfAggregates(
+              "SELECT BIT_AND(col1), BIT_OR(col1), BIT_XOR(col1), MIN(col1), COUNT(col1) FROM test",
+              expectedNumOfCometAggregates)
+
+            checkSparkAnswerAndNumOfAggregates(
+              "SELECT BIT_AND(col1), BIT_OR(col1), BIT_XOR(col1), col3 FROM test GROUP BY col3",
+              expectedNumOfCometAggregates)
+
+            checkSparkAnswerAndNumOfAggregates(
```
[the remark this answers is not in the document]
Comment (huaxingao) [reply spark/src/test/scala/org/apache/comet/exec/CometAggregateSuite.scala:987]: Comments were added. Thanks

File: spark/src/main/scala/org/apache/comet/serde/QueryPlanSerde.scala [hunk 3]
```diff
@@ -186,6 +186,13 @@ object QueryPlanSerde extends Logging with ShimQueryPlanSerde {
     }
   }
 
+  private def bitwiseAggTypeSupported(dt: DataType): Boolean = {
+    dt match {
+      case _: IntegerType | LongType | ShortType | ByteType => true
+      case _ => false
+    }
+  }
```
Comment (viirya) [inline spark/src/main/scala/org/apache/comet/serde/QueryPlanSerde.scala:194]: Is the limitation due to DataFusion?

Comment (viirya) [reply spark/src/main/scala/org/apache/comet/serde/QueryPlanSerde.scala:194]: If Spark bit aggregate function supports more types than DataFusion, can you add a comment here and open a ticket at DataFusion?

**Review by viirya — APPROVED** [review]

File: spark/src/main/scala/org/apache/comet/serde/QueryPlanSerde.scala (hunk 3 above)
Comment (huaxingao) [reply spark/src/main/scala/org/apache/comet/serde/QueryPlanSerde.scala:194]: Spark bitwise aggregates only supports `INTEGRAL` type. If I try other types, Spark throws Exception:
```
Cannot resolve "bit_and(col1)" due to data type mismatch: Parameter 1 requires the "INTEGRAL" type, however "col1" has the type ...
```

**Review by comphead — APPROVED** [review]
lgtm thanks @huaxingao

**Comment by sunchao** [comment]
Merged, thanks!

**Comment by huaxingao** [comment]
Thank you all!

**Pull request merged**